force-app/main/default/classes/AccountTriggerHandlerTest.cls

```diff
@@ -0,0 +1,36 @@
+/**
+ * description  Test Class for AccountTriggerHandler
+ * @author      Egor Apaniuk
+ * @since       31/01/2024
+ */
+@IsTest
+public with sharing class AccountTriggerHandlerTest {
```

Comment:
private: No outside class can access the class, but you can run test methods on this class
нужен ли здесь public?

почему выбрал именно with sharing?
https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_best_practices.htm
https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_testing_tools_runas.htm
Generally, all Apex code runs in system mode, where the permissions and record sharing of the current user aren’t taken into account. The system method runAs enables you to write test methods that change the user context to an existing user or a new user so that the user’s record sharing is enforced. The runAs method enforces record sharing.
Use the runAs method to test your application in different user contexts - best practice